scripts/public_api_report.py

```diff
@@ -784,6 +854,394 @@ class CodeReferenceFilter:
+            filepath=pathlib.Path("great_expectations/dataset/sparkdf_dataset.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            name="expect_column_values_to_not_be_null",
+            filepath=pathlib.Path("great_expectations/dataset/sparkdf_dataset.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            name="head",
+            filepath=pathlib.Path("great_expectations/dataset/sparkdf_dataset.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            name="expect_column_values_to_be_between",
+            filepath=pathlib.Path("great_expectations/dataset/sqlalchemy_dataset.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            name="expect_column_values_to_not_be_null",
+            filepath=pathlib.Path("great_expectations/dataset/sqlalchemy_dataset.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="Exclude code from v2 API",
+            name="head",
+            filepath=pathlib.Path("great_expectations/dataset/sqlalchemy_dataset.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="self_check is mentioned but in the docs we currently recommend using test_yaml_config which uses self_check under the hood. E.g. https://docs.greatexpectations.io/docs/guides/setup/configuring_data_contexts/how_to_configure_datacontext_components_using_test_yaml_config/#steps",
+            name="self_check",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/data_connector/data_connector.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="self_check is mentioned but in the docs we currently recommend using test_yaml_config which uses self_check under the hood. E.g. https://docs.greatexpectations.io/docs/guides/setup/configuring_data_contexts/how_to_configure_datacontext_components_using_test_yaml_config/#steps",
+            name="self_check",
+            filepath=pathlib.Path(
+                "great_expectations/datasource/data_connector/runtime_data_connector.py"
+            ),
+        ),
+        IncludeExcludeDefinition(
+            reason="self_check is mentioned but in the docs we currently recommend using test_yaml_config which uses self_check under the hood. E.g. https://docs.greatexpectations.io/docs/guides/setup/configuring_data_contexts/how_to_configure_datacontext_components_using_test_yaml_config/#steps",
+            name="self_check",
+            filepath=pathlib.Path("great_expectations/datasource/new_datasource.py"),
+        ),
+        IncludeExcludeDefinition(
+            reason="False match for dict `.update()` method.",
+            name="update",
+            filepath=pathlib.Path(
+                "great_expectations/execution_engine/execution_engine.py"
+            ),
+        ),
```

Comment:
Would this be considered a basic CRUD method for Execution Engine, and so part of the Public API? Or do we not need that for the Execution Engine class.

Reply:
Good question - this method doesn't exist though on the Execution Engine. It is an artifact of the string matching we are doing to find usages in our documentation examples.